broke-broker/src/main/java/org/dsngroup/broke/broker/storage/InMemoryPool.java

```diff
@@ -27,16 +28,24 @@
     // TODO: Assignable pool size
     // TODO: Extends this to be offset based data structure.
     // TODO: May consider to replace ConcurrentHashMap into more performant data structure.
-    private static Map<String, String> inMemoryPool = new ConcurrentHashMap<>();
+    private static Map<String, ArrayList<String>> inMemoryPool = new ConcurrentHashMap<>();
 
     /**
      * Insert storage on a specific topic.
      * @param topic the topic(key) of the broker.
      * @param content the storage of the associated topic.
      */
     public static void putContentOnTopic(String topic, String content) {
+
+        // Initialize the array list for the first-time topic
+        if (inMemoryPool.get(topic) == null)
+            inMemoryPool.put(topic, new ArrayList());
+
         // Ignore the return value
-        inMemoryPool.put(topic, content);
+        inMemoryPool.get(topic).add(content);
```

Comment:
In my guess, the `ArraList` here is used for concat the content under associated topic?
Try for offset or partition?

Reply:
Yes. And ArrayList can be an efficient data structure for continuous data accessing.

Reply:
I think we can discuss on this later b.c. the opinion of mine under.